lib/block-template-utils.php

```diff
@@ -95,9 +96,52 @@ function gutenberg_generate_block_templates_export_file() {
 		$theme_json_raw     = array_merge( $schema, $theme_json_raw );
 	}
 
+	// Find any uploaded files.
+	$uris_to_migrate = WP_Theme_JSON_Resolver_Gutenberg::get_migrated_relative_theme_uris(
+		$tree,
+		array(
+			'relative_path_prefix' => 'file:./assets/',
+		)
+	);
+	if ( ! empty( $uris_to_migrate ) ) {
+		$uploads = wp_upload_dir();
+		foreach ( $uris_to_migrate as $uri ) {
+			$href   = $uri['href'];
+			$target = $uri['target'];
+			if ( str_ends_with( $target, 'background.backgroundImage.url' ) ) {
+				/*
+				 * For background images, reset the backgroundImage object
+				 * to remove upload "id", "source", and "title".
+				 * Done by removing .url from the path to get the target, and setting
+				 * href to replace the `background.backgroundImage` object.
+				 */
+				$target = rtrim( $target, '.url' );
+				$href   = array(
+					'url' => $href,
+				);
+			}
```

Comment:
I can test this in the Core patch since Core tests have access to ZipArchive. 

E.g., https://github.com/WordPress/wordpress-develop/blob/de38e8eeecf80a7ca5ee870ec68a59897216b33b/tests/phpunit/tests/block-template-utils.php#L379